src/components/fx/hover.js

```diff
@@ -333,7 +333,12 @@ function _hover(gd, evt, subplot, noHoverEvent) {
                 return;
             }
 
-            var dbb = evt.target.getBoundingClientRect();
+            // Discover event target, traversing open shadow roots.
+            var target = evt.composedPath && evt.composedPath()[0];
```

Comment:
Have you also investigated a possible workaround for IE?

Reply:
I haven't -- I'll do so and update with my findings.

Reply:
After some investigation, I think the best we can do in the case of IE is simply fallback to the original behavior of directly using the event target.

```suggestion
            var target = evt.composedPath && evt.composedPath()[0];
            if(!target) {
                target = evt.target;
            }
```

Reply:
Thanks. Please commit your suggestion.

Reply:
Or simply
```js
var target = (evt.composedPath && evt.composedPath()[0]) || evt.target;
```
@dbluhm in addition to hover (fx), there are some other places e.g. `dragelement`, `shapes` and `mapbox` which `evt.target` is used. Could you please investigate if there is a issue that similar fix may help?

Reply:
ping @dbluhm

Reply:
Sure, I'll take a look! Apologies for the slow response -- working to address this issue just as time arises. Thanks for your patience!

Reply:
Thanks @dbluhm - this looks like a great solution! My hunch is that in order to generalize this efficiently to all the places we use `event.target` it will warrant a function like `Lib.getTarget(event)`, perhaps in [lib/dom](https://github.com/plotly/plotly.js/blob/master/src/lib/dom.js) (and note that then this needs to be exposed in [lib/index](https://github.com/plotly/plotly.js/blob/5700db226a518d9eadcf8f35ffb29445f91e7ff3/src/lib/index.js#L144))